R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,456 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      L2322.GlobalTechCoef_Fert
+
+    # Costs of global technologies
+    # L2322.GlobalTechCost_Fert: Non-energy costs of global fertilizer manufacturing technologies
+    L2322.GlobalTechCoef_Fert %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechYr"]]) %>%
+      mutate(minicam.non.energy.input = "non-energy") %>%
+      left_join(L1322.Fert_NEcost_75USDkgN_F, by = c('technology' = 'fuel') ) %>%
+      rename(input.cost = NEcost_75USDkgN) %>%
+      mutate(input.cost = round(input.cost, energy.DIGITS_COST)) %>%
+      na.omit -> #Export technologies have no cost assigned. Just drop the object
+      L2322.GlobalTechCost_Fert
+
+    # Carbon capture rates from technologies with CCS
+    # L2322.GlobalTechCapture_Fert: CO2 capture fractions from global fertilizer production technologies with CCS
+    ## No need to consider historical periods or intermittent technologies here
+    A322.globaltech_co2capture %>%
+      gather(year, remove.fraction, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_co2capture_long
+
+    df_years <- unique(A322.globaltech_co2capture_long$year)
+    int_years <- setdiff(FUTURE_YEARS, df_years)
 
+    A322.globaltech_co2capture_long %>%
+      select(-remove.fraction, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_co2capture_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(remove.fraction = approx_fun(year, remove.fraction, rule = 1)) %>%
+      mutate(remove.fraction = round(remove.fraction, energy.DIGITS_REMOVE.FRACTION)) %>%
+      ungroup %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(storage.market = "carbon-storage") ->
+      L2322.GlobalTechCapture_Fert
+
+    # Retirement information
+    A322.globaltech_retirement %>%
+      set_years %>%
+      mutate(year = as.numeric(year)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      A322.globaltech_retirement_with_years
+
+    # Copy the data in the first future period through to the end year
+    A322.globaltech_retirement_with_years %>%
+      filter(year == max(BASE_YEARS)) ->
+      A322.globaltech_retirement_max_baseyear
+
+    A322.globaltech_retirement_with_years %>%
+      filter(year == min(FUTURE_YEARS)) %>%
+      select(-year) %>%
+      repeat_add_columns(tibble(year = FUTURE_YEARS)) %>%
+      bind_rows(A322.globaltech_retirement_max_baseyear) ->
+      L2322.globaltech_retirement
+
+    # Retirement may consist of any of three types of retirement function (phased, s-curve, or none)
+    # All of these options have different headers, and all are allowed
+
+    # L2322.GlobalTechShutdown_Fert: Global tech lifetime and shutdown rate
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
```

Comment:
Again, if they contain no data and aren't being tested, why are we generating these?

Reply:
Might be nbd but this might also be an issue more broadly...the old data system had lots of level2 data files that weren't written out in the core assumptions, but if users modified certain assumptions, then they would be generated. We might be losing some capabilities there, since those files that aren't generated for the core set of assumptions aren't being tested.

Reply:
All the empty/no-test data tables generated in this chunk were conditional outputs in the old chunk. Creating an empty/no-test data table for conditional output is how we deal with such situation as discussed in #633

Reply:
This is different from #633; in that case, the issue was that the old code was referencing a non-existent CSV, and it only didn't crash because it was within a conditional statement that (in testing and in application) was always FALSE. So the matter discussed in #633 was whether to add functionality that wasn't there to begin with, and we decided not to because it was not useful.
The difference here is that the present approach is dropping existing functionality. Where users can currently set share-weight paths year-by-year (in the "year" column of A322.subsector_shrwt), and choose between different retirement functions (s-curve and phased-shutdown in A322.globaltech_retirement), that capability is no longer supported in this present approach. That might be OK for this particular sector, and that's a different discussion, but I agree with @bpbond that if we do decide to ditch this capability, there is no point in making x=NA tables.

Reply:
Thanks for your note! @pkyle

So how do we decide for those empty tables? Remove them all? @bpbond

Reply:
I removed the empty tibbles for now unless we decide to do something else.